core/src/main/java/io/grpc/transport/AbstractClientStream.java

```diff
@@ -45,9 +47,12 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import javax.annotation.concurrent.NotThreadSafe;
+
 /**
  * The abstract base class for {@link ClientStream} implementations.
  */
+@NotThreadSafe
```

Comment:
@carl-mastrangelo, I don't think you replied to my concern. How is this class different from any other non-thread-safe class (which is defined by "a class that did not specify it was thread-safe)? Why does _this_ class need it? I reject the idea of putting NotThreadSafe on every class, and so I want to make sure that it is clear that there should be a specific reason why it is added in each instance.

(I can even come up with some reasons myself, but I don't want to sign up for having the NotThreadSafes everywhere)